invenio_app_ils/records/views.py

```diff
@@ -9,28 +9,112 @@
-    DOCUMENT_REQUEST_PID_TYPE, EITEM_PID_TYPE
+    DOCUMENT_REQUEST_PID_TYPE, EITEM_PID_TYPE, PATRON_PID_TYPE
 from invenio_app_ils.proxies import current_app_ils
+from invenio_app_ils.records.serializers.access import patron_list_serializer
 from invenio_app_ils.signals import record_viewed
 
 
+def create_access_list_blueprint(app):
+    """Create access list blueprint."""
+    blueprint = Blueprint("ils_access_list", __name__, url_prefix="")
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+
+    # Get the list not the document
+    options = endpoints.get(PATRON_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    serializers = {"application/json": patron_list_serializer}
+    access_list_view = AccessListResource.as_view(
+        AccessListResource.view_name,
+        serializers=serializers,
+        default_media_type=default_media_type,
+    )
+    blueprint.add_url_rule(
+        "/access/list",
+        view_func=access_list_view,
+        methods=["GET"],
+    )
+    return blueprint
+
+
+class AccessListResource(ContentNegotiatedMethodView):
+    """Access list endpoint."""
+
+    view_name = "access_list"
+
+    @use_args({"q": fields.Str(missing='')})
+    @need_permissions("document-access-update")
+    def get(self, args, **kwargs):
+        """Get the list of avaialabe options for restrictions."""
+        result = current_app.config["ILS_ACCESS_LIST_SEARCH"](args["q"])
+        return self.make_response(
+            patrons=result,
+            total=result.hits.total.value,
+            code=200
+        )
+
+
+def create_document_access_blueprint(app):
+    """Create document update access blueprint."""
+    blueprint = Blueprint("ils_document_access", __name__, url_prefix="")
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+    options = endpoints.get(DOCUMENT_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    rec_serializers = options.get("record_serializers", {})
+    serializers = {
+        mime: obj_or_import_string(func)
+        for mime, func in rec_serializers.items()
+    }
+
+    access_views = DocumentAccessResource.as_view(
+        DocumentAccessResource.view_name.format(DOCUMENT_PID_TYPE),
+        serializers=serializers,
+        default_media_type=default_media_type,
+    )
+    blueprint.add_url_rule(
+        "{0}/access".format(options["item_route"]),
+        view_func=access_views,
+        methods=["POST"],
+    )
+    return blueprint
+
+
+class DocumentAccessResource(ContentNegotiatedMethodView):
+    """Document endpoint to update access list."""
+
+    view_name = "{}_access"
+
+    @need_permissions("document-access-update")
+    @pass_record
+    def post(self, pid, record, **kwargs):
+        """Update document access list."""
+        data = request.get_json()
+        access = record.get("_access", {})
+        for key, val in data.items():
+            access[key] = val
```

Comment:
You are trusting too much from the client here, I thing there should be more validation.